src/hotspot/share/adlc/output_c.cpp

```diff
@@ -4010,6 +4014,22 @@ void ArchDesc::buildMachNode(FILE *fp_cpp, InstructForm *inst, const char *inden
     fprintf(fp_cpp, " );\n");
     // #####
   }
+  if (inst->_flag != nullptr) {
+    Flag* node = inst->_flag;
+    const char* prefix = "Node::";
+    bool node_flags_set = false;
+    do {
+      if (!node_flags_set) {
+        fprintf(fp_cpp, "%s node->add_flag(%s%s", indent, strncmp(node->_name, prefix, strlen(prefix)) != 0 ? prefix : "", node->_name);
```

Comment:
This seems to be guarding against a case where the flag is declared with the prefix already in the .ad file. Is this required for something?

(Otherwise I suggest just using `node->_name` here, as it forces the flag declarations in the .ad file to be consistent).

Reply:
No, this is just a leftover from an earlier design, I'll remove the check and always add the prefix